apps/cloud_inference/server_app/source/inference_hip_rocal.cpp

```diff
@@ -0,0 +1,1092 @@
+                            if(prob > max_prob) {
+                                label = c;
+                                max_prob = prob;
+                            }
+                        }
+                    }
+                    outputQ.enqueue(std::tuple<int,int>(tag,label));
+                }else {
+                    // todo:: add support for fp16
+                    std::vector<float>  prob_vec((float*)buf, (float*)buf + dimOutput[2]);
+                    std::vector<size_t> idx(prob_vec.size());
+                    std::iota(idx.begin(), idx.end(), 0);
+                    sort_indexes(prob_vec, idx);            // sort indeces based on prob
+                    std::vector<unsigned int>    labels;
+                    outputQ.enqueue(std::tuple<int,int>(tag,idx[0]));
+                    int j=0;
+                    for (auto i: idx) {
+                        // make label which is index and prob
+                        int packed_label_prob = (i&0xFFFF)|(((unsigned int)((prob_vec[i]*0x7FFF)+0.5))<<16);   // convert prob to 16bit float and store in MSBs
+                        labels.push_back(packed_label_prob);
+                        if (++j >= topK) break;
+                    }
+                    outputQTopk.enqueue(labels);
+                }
+            }else
+            {
+                std::vector<ObjectBB> detected_objects;
+                region->GetObjectDetections((float *)buf, BB_biases, dimOutput[2], dimOutput[1], dimOutput[0], BOUNDING_BOX_NUMBER_OF_CLASSES, dimInput[0], dimInput[1], BOUNDING_BOX_CONFIDENCE_THRESHHOLD, BOUNDING_BOX_NMS_THRESHHOLD, 13, detected_objects);
+                if (detected_objects.size() > 0) {
+                    // add it to outputQ
+                    outputQ.enqueue(std::tuple<int,int>(tag,detected_objects[0].label));
+                    // add detected objects with BB into BoundingBox Q
+                    OutputQBB.enqueue(detected_objects);
+                } else
+                {
+                    // add it to outputQ
+                    outputQ.enqueue(std::tuple<int,int>(tag,-1));
+                }
+            }
+        }
+
+        // add the output back to idle queue
+        queueDeviceOutputMemIdle[gpu]->enqueue(output);
+
+        PROFILER_STOP(inference_server_app, workDeviceOutputCopy);
+
+        // update counter
+        if(outputCount > 0) {
+            totalBatchCounter++;
+            totalImageCounter += outputCount;
+        }
+    }
+
+    // release HIP stream
+    hipStreamDestroy(stream);
+
+    // send end of sequence marker to next stage
+    outputQ.enqueue(std::tuple<int,int>(-1,-1));
+    args->lock();
+    info("workDeviceOutputCopy: GPU#%d terminated for %s [processed %d batches, %d images]", gpu, clientName.c_str(), totalBatchCounter, totalImageCounter);
+    args->unlock();
+}
+#endif
+
+void InferenceEngineRocalHip::dumpBuffer(hipStream_t stream, void * mem, size_t size, std::string fileName)
```

Comment:
Already defined in base class. do not redefine here

Reply:
Removed